### DISCUSSION TITLE
Window snap

### DISCUSSION THREAD

**Discussion by probonopd** [WayfireWM/wayfire#2018] [Q&A] [answered]
Is it possible to have window snap, where the window e.g., takes the left half of the screen when you drag it onto the left border of the screen?

**Comment by soreau** [comment]
Yes, I think you need move and grid plugins enabled.

**Reply by probonopd** [reply]
How would I do this? Raspberry Pi OS doesn't seem to have a GUI for configuring Wayfire.

**Reply by probonopd** [reply]
As soon as I add

```
[core]
plugins = move grid
```

and restart Wayfire, then I see only a dark background with a mouse cursor but the desktop does not appear.

Raspberry Pi OS ships `0.7.5-1~bpo11+1~rpt28`.

**Reply by soreau** [reply]
You would want move and grid in addition to other plugins that make the desktop environment function, such as autostart. You can see a default config file [here](https://github.com/WayfireWM/wayfire/blob/0.7.x/wayfire.ini). However without any config file, [this](https://github.com/WayfireWM/wayfire/blob/0.7.x/metadata/core.xml#L10) plugin set will be loaded anyway, so not having a config file is probably ok too. There are some options that might need to be enabled too, namely [this](https://github.com/WayfireWM/wayfire/blob/0.7.x/metadata/move.xml#L13).

**Reply by soreau** [reply]
Looks like they do this as well in their packaged version of wayfire https://github.com/raspberrypi-ui/wayfire/blob/master/debian/patches/defaultconf.patch

**Comment by probonopd** [comment, marked as answer]
Apparently my `.config/wayfire.ini` was messed up.

After doing `cp /etc/wayfire/defaults.ini .config/wayfire.ini` and editing it to set `mouse_snap = true` (it defaults to false for whatever reason) it is working now. Thanks for the help!
